### PR TITLE
epsonscan2: 6.7.63.0 -> 6.7.70.0

### PR DESCRIPTION
**Pull request by Shawn8901**
Updates epsonscan2 to the latest version. As flathub is not shipping that version is switched patch source from flathub to AUR.
Aside that i updated our build.patch and added a gcc14 patch (as i noticed that epsonscan2 does not build on staging-next).
In addition the icon is now showing correctly when built with `withGUI = true;` 

For testing a scanned a page with a ET-3750 using epsonscan2 GUI.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/misc/drivers/epsonscan2/default.nix [hunk 1]
```diff
@@ -55,26 +55,29 @@ stdenv.mkDerivation {
...
     (fetchpatch {
-      url = "https://raw.githubusercontent.com/flathub/net.epson.epsonscan2/master/patches/epsonscan2-xdg-open.patch";
-      hash = "sha256-4ih3vZjPwWiiAxKfpLIwbbsk1K2oXSuxGbT5PVwfUsc=";
+      url = "https://aur.archlinux.org/cgit/aur.git/plain/0003-Use-XDG-open-to-open-the-directory.patch?h=epsonscan2";
+      hash = "sha256-H3lle1SXkkpbBkozYEwiX0z9oTUubTpB+l91utxH03M=";
     })
   ];
 
   postPatch = ''
+    rm CMakeCache.txt
```
Comment (Shawn8901) [inline pkgs/misc/drivers/epsonscan2/default.nix:74]: fyi that is needed as the currents version archive contains that cmake cache file, thus its unhappy about our build system.

File: pkgs/misc/drivers/epsonscan2/default.nix [hunk 2]
```diff
@@ -127,8 +130,8 @@ stdenv.mkDerivation {
     + lib.optionalString withGui ''
       # The icon file extension is .ico but it's actually a png!
       mkdir -p $out/share/icons/hicolor/{48x48,128x128}/apps
-      convert $src/Resources/Icons/escan2_app.ico -resize 48x48 $out/share/icons/hicolor/48x48/apps/epsonscan2.png
-      convert $src/Resources/Icons/escan2_app.ico -resize 128x128 $out/share/icons/hicolor/128x128/apps/epsonscan2.png
+      convert $src/Resources/Icons/escan2_app.ico -resize 48x48 -delete 1,2,3 $out/share/icons/hicolor/48x48/apps/epsonscan2.png
+      convert $src/Resources/Icons/escan2_app.ico -resize 128x128 -delete 1,2,3 $out/share/icons/hicolor/128x128/apps/epsonscan2.png
```
Comment (Shawn8901) [inline pkgs/misc/drivers/epsonscan2/default.nix:134]: icons havent worked, as the ico contains multiple resolutions and resulting in multiple pngs after convertion and thus not being named correctly

**Review by james-atkins — APPROVED** [review]

**Comment by Shawn8901** [comment]
> Thanks! I pushed PR #358303 a while ago but it hasn't been merged and so this would supersede it. Nice catch with the icons.

ah, i did not see that pr, i was mainly motivated by the compile error with gcc14. Should i split that out so that we can cherry-pick those to your pr?

File: pkgs/misc/drivers/epsonscan2/default.nix [hunk 3]
```diff
@@ -55,26 +55,29 @@ stdenv.mkDerivation {
 
   patches = [
     ./build.patch
+    ./gcc14.patch
     (fetchpatch {
-      url = "https://github.com/flathub/net.epson.epsonscan2/raw/master/patches/epsonscan2-crash.patch";
-      hash = "sha256-srMxlFfnZuJ3ed5veFcJIiZuW27F/3xOS0yr4ywn4FI=";
+      url = "https://aur.archlinux.org/cgit/aur.git/plain/0002-Fix-crash.patch?h=epsonscan2";
```
Comment (FliegendeWurst) [inline pkgs/misc/drivers/epsonscan2/default.nix:60]:
```suggestion
      url = "https://aur.archlinux.org/cgit/aur.git/plain/0002-Fix-crash.patch?h=epsonscan2&&id=6e422c487f5f3b135df554fde65ab57bc6377159";
```
Same for other URLs, to ensure the content doesn't change.

Comment (Shawn8901) [reply pkgs/misc/drivers/epsonscan2/default.nix]: i did so, but by keeping the flatpak patch sources, as they also apply correctly on 6.7.70.0 (which i did not even check before).

**Comment by Shawn8901** [comment]
as staging-next was merged. That becomes an unbreak PR for master (already targets master).

**Review by james-atkins — APPROVED** [review]

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**